lineage/UpwardRecursion.py

```diff
@@ -201,3 +201,5 @@ def get_nonleaf_NF_and_betas(tHMMobj, MSD: list, EL: list, NF: list, betas: list
 
                 NF[num][pii] = sum(fac1)
                 betas[num][pii, :] = fac1 / NF[num][pii]
+
+    return betas, NF
```

Comment:
This is one of the places where the function was not returning the updated matrices.
The other is `deltas` from `Viterbi()`